### DISCUSSION TITLE
Using ILI9488 with LVGL on ESP32

### DISCUSSION THREAD

**Discussion by PWSJA** [micropython/micropython#10346] [Libraries & Drivers]
MicroPython-version: v1.19.1, downloaded from https://micropython.org/download/esp32/.
Board: ESP32 DevkitC V4
We work with VSCode and the Pymakr extension.

Hello everyone,
We are quite new to MicroPython and are trying to display something on an ILI9488 using LVGL. Our ESP32 works fine. We have fixed it to display something on a segment display, but now we are trying to display something on a TFT-display.
We have tried to find a clear tutorial about which libraries we need and how to install them to get it working, but we got quite confused about:
-	The libraries we need (which lvgl library do we need)
-	How to install these libraries on our ESP32.

Can anyone explain us how we can do this, please?

**Comment by davefes** [comment]
https://github.com/lvgl/lv_micropython
https://github.com/Bodmer/TFT_eSPI/issues/187

Maybe they help.

**Reply by PWSJA** [reply]
Firstly thank you for your message! We tried the steps in your first link, but we got some errors.  The first error we got was after we typed 
```py
make -C mpy-cross
```
```py
make: Entering directory 'E:/Software/Git/lv_micropython/mpy-cross'
Use make V=1 or set BUILD_VERBOSE in your environment to increase build verbosity.
find: paths must precede expression: `mphalport.h'
find: possible unquoted pattern after predicate `-name'?
process_begin: CreateProcess(NULL, gcc -dumpmachine, ...) failed.
Makefile:54: pipe: No error
process_begin: CreateProcess(NULL, gcc -dumpmachine, ...) failed.
../py/mkrules.mk:177: pipe: Bad file descriptor
LVGL-GEN build/lvgl/lv_mpy.c
/usr/bin/sh: line 1: gcc: command not found
make: *** [../py/py.mk:46: build/lvgl/lv_mpy.c] Error 127
make: Leaving directory 'E:/Software/Git/lv_micropython/mpy-cross'
```
After this command, we typed the following command: 
```py
make -C ports/esp32 BOARD=GENERIC deploy
```
This didn't work and gave the following errors:
```py
make: Entering directory 'E:/Software/Git/lv_micropython/ports/esp32'
idf.py -D MICROPY_BOARD=GENERIC -B build-GENERIC  -p /dev/ttyUSB0 -b 460800 flash
process_begin: CreateProcess(NULL, idf.py -D MICROPY_BOARD=GENERIC -B build-GENERIC -p /dev/ttyUSB0 -b 460800 flash, ...) failed.
make (e=2): Het systeem kan het opgegeven bestand niet vinden.
make: *** [Makefile:49: deploy] Error 2
make: Leaving directory 'E:/Software/Git/lv_micropython/ports/esp32'
```
We couldn't get this working, although we followed the steps. Do you have an idea why this error occurs?

**Reply by davefes** [reply]
I haven't used make for years.  I think there has been a major change as I hear the word cmake used for the ESP32, instead of make.  Sorry, needs someone with relevant experience.
